src/classes/WalletConnectors/WalletConnect.ts

```
@@ -0,0 +1,130 @@
+    projectId: string
+    /**
+     * Metadata for your app.
+     * @link https://docs.walletconnect.com/2.0/advanced/providers/ethereum#initialization
+     */
+    metadata?: EthereumProviderOptions['metadata']
+    /**
+     * Whether or not to show the QR code modal.
+     * @default true
+     * @link https://docs.walletconnect.com/2.0/advanced/providers/ethereum#initialization
+     */
+    showQrModal?: boolean
+    /**
+     * Options of QR code modal.
+     * @link https://docs.walletconnect.com/2.0/web/walletConnectModal/modal/options
+     */
+    qrModalOptions?: EthereumProviderOptions['qrModalOptions']
+    /**
+     * Option to override default relay url.
+     * @link https://docs.walletconnect.com/2.0/web/providers/ethereum
+     */
+    relayUrl?: string
+}
+
+export class WalletConnect extends WalletConnector {
+    public readonly id: string = 'walletConnect'
+    public readonly name: string = 'Wallet Connect'
+
+    private readonly _options: Omit<WalletConnectOptions, 'icon'>
+    private _provider?: WalletConnectProvider
+
+    constructor({ icon, ...options }: WalletConnectOptions) {
+        super({ icon: icon ?? icons.walletconnect })
+
+        this._options = options
+    }
+
+    public async getProvider() {
+        const { EthereumProvider } = await import('@walletconnect/ethereum-provider')
+
+        const {
+            chains,
+            projectId,
+            metadata,
+            qrModalOptions,
+            relayUrl,
+            showQrModal = true,
+        } = this._options
+
+        const [defaultChain, ...optionalChains] = chains
+
+        this._provider = await EthereumProvider.init({
+            projectId,
+            chains: [defaultChain],
+            optionalChains,
+            showQrModal,
+            metadata,
+            qrModalOptions,
+            relayUrl,
+        })
+
+        try {
+            await this._provider.connect()
+
+            return this._provider
+        }
+        catch { }
```

empty catch isnt ideal